### PR TITLE
Support scraper logging to specific log levels

### LINKED ISSUE #1348: [Feature] Support Plugin like logging with Scrapers
**Is your feature request related to a problem? Please describe.**
Scrappers can currently only log to `ErrorLevel` anything sent to stderr is [currently sent to the error log](https://github.com/stashapp/stash/blob/70b66d91a07b9166707b196db84e86362dd14927/pkg/scraper/script.go#L72)

**Describe the solution you'd like**
Allow [plugin like](https://github.com/stashapp/stash/blob/70b66d91a07b9166707b196db84e86362dd14927/pkg/plugin/log.go#L12) logging from scrapers

**Describe alternatives you've considered**
Logging exclusively to error log, this is usable, it was just initially confusing that logging for the scraper didn't work like plugins did

### PR DESCRIPTION
**Pull request by stg-annon**
Resolves #1348

I don't have any experience with Go most of this was pulled directly from parts of how plugins handle logging

### CODE REVIEW

File: pkg/scraper/script.go [hunk 1]
```diff
@@ -207,3 +203,66 @@ func findPythonExecutable() (string, error) {
...
+		// 	} else {
+		// 		// only pass progress through if channel present
+		// 		if s.progress != nil {
+		// 			// don't block on this
+		// 			select {
+		// 			case s.progress <- progress:
+		// 			default:
+		// 			}
+		// 		}
+		// 	}
```
Comment (bnkai) [inline pkg/scraper/script.go:244]: Progress shouldnt be needed when scraping and would be more complex to implement. Just delete the comments imo

**Comment by bnkai** [comment]
It seems to work ok.
A modified IAFD scraper to test it with can be found [here](https://raw.githubusercontent.com/bnkai/CommunityScrapers/iafd_logging/scrapers/IAFD.py)

**Comment by WithoutPants** [comment]
I have refactored the plugin logging code to move it into a common area to minimise code duplication. Functionally it should be the same, but we'll need regression testing of the plugin logging functionality.

**Comment by bnkai** [comment]
Looks and works ok.
Tested against an `old` script scraper the IAFD updated one mentioned above and the GHScraper_Checker plugin.

**Review by bnkai — APPROVED** [review]

**Comment by stg-annon** [comment]
I think there may be an issue with logging to Web UI log window, logs seem to be coming through the terminal window fine though

**Comment by gitgiggety** [comment]
I can confirm the comment of @stg-annon above.
Took me way too long to stop trial & error to get the log shown, and continue to read all comments and find out they indeed are shown in the terminal, just not in the web UI.

(And if fixes / improvements have to be made, maybe include the scraper name as well? Something like `[Scraper / <name>] <the message>` instead of just `[Scraper]`)

**Pull request merged**